Automattic/wp-calypso
# State: Migrate lib/post-formats to Redux state

This PR migrates the legacy `lib/post-formats` Flux store to Redux. It fixes #7925 and is part of the group effort to use a single data tree and query components throughout Calypso - #5046.

This PR introduces:

- [x] Action creators
- [x] Reducer
- [x] Selectors
- [x] Schema
- [x] Query components - `QueryPostFormats`

Tests are also included.

To test:

* Run `npm run test-client client/state/post-formats` and verify all tests pass ✅ 

Implementation of this and removal of the old Flux store will be done in the subsequent PR(s).

/cc @aduth for review
